libavpipe/src/avpipe_xc.c

```diff
@@ -4210,6 +4210,8 @@ avpipe_probe(
 
 avpipe_probe_end:
     if (decoder_ctx.format_context) {
+        if (decoder_ctx.format_context->pb->buffer)
```

Comment:
There is one more piece of memory that can be freed:
```
            AVIOContext *avioctx = (AVIOContext *) decoder_ctx.format_context->pb;
            if (avioctx) {
                av_freep(&avioctx->buffer);
                av_freep(&avioctx);
            }

```

Reply:
agreed and changed